pkg/agent/flowexporter/connections/connections.go

```diff
@@ -97,7 +99,25 @@ func (cs *connectionStore) getConnByKey(flowTuple flowexporter.ConnectionKey) (*
 	return &conn, found
 }
 
+func (cs *connectionStore) IterateCxnMapWithCB(updateCallback flowexporter.FlowRecordUpdate) error {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+
+	for k, v := range cs.connections {
+		cs.mutex.Unlock()
```

Comment:
Not safe to iterate a map without lock? What happens if another routine add/delete keys?

Reply:
I want to provide some further clarification, by considering a scenario, w.r.t following code. Here as well if routine executing iterate grabs the lock first, then we will build records using old connection data. However, by releasing lock after every item, we can decrease the contention. 

lock 
addOrUpdateConnectionMap
unlock

lock
IterateConnectionMap
unlock

Also, I would like to note that range connectionMap will iterate on old copy of connection map even connection map is updated in other routine. I missed that in my previous comment. Thanks.

Reply:
I did not know map iteration in Golang is thread safe.
Anyway could you add some comments to explain why you release the lock there, and what could be the impact?

Reply:
You are right, Jianjun. Iteration over the map is not thread-safe depending on concurrent operations. Releasing lock in our case may work because of current deletion logic, and the presumption that sending old connection data is fine. Added the comment in the code. Please take a look.

Reply:
This sounds tricky. Have you verified it really works?

Reply:
It worked as expected with respect to deletion logic. However, because of the IPFIX collector requirement of sending accurate flow records rather than from the last poll cycle, I changed the synchronization logic.